Several small css changes

roberthouse54 opened the pull request:
#### What's this PR do?

This will be a series of changes to the css. Included in this PR: changed the background color on the body, removed the bgcolor on page-content. Removed the top/bottom margins on the layout columns, removed the bottom-padding on the navbar. 

#### Where should the reviewer start?

Look at the changes in the css files and see that nothing looks broken.

#### Any background context you want to provide?

This is not related to a specific issue, but I will be making css changes on on ongoing basis, doing a lot of tweaks... some small visual refinements, and some larger layout changes.

aliceriot commented on static/scss/general_styles.scss:
```diff
@@ -0,0 +1,15 @@
+
+
+
+
```
could you delete the white space here?

aliceriot replied:
additionally, as George pointed out above css should generally be written like this:

```scss
selector {
  rule: value;
}
```

so that the closing brace is at the same indentation level as the first character in the selector, each rule is on it's own line, and any indentation is by 2 spaces at a time. This means a nested dealio would look like:

```scss
.foo {
  .bar {
    text-align: justify;
  }

  .baz {
    color: purple;
  }
}
```

I prefer to leave a blank line after a block (i.e. after the closing brace for the `.bar` block} unless I'm closing multiple blocks at once, in which case I think the braces should cascade (as they do after the `.baz` block).

let me know if that doesn't make sense, haha

aliceriot replied:
I think george said something about configuring sublime, but all this is something that your editor can be configured to mostly do for you.

aliceriot commented on static/scss/homepage.scss:
```diff
@@ -45,6 +45,9 @@ body.app-media {
 }
+
+
+
```
here too :)

aliceriot commented:
the spacing on the profile is a little wacky now:

![profile_spacing](https://cloud.githubusercontent.com/assets/6207644/17789264/635c71cc-655f-11e6-81b6-de8c125b62be.png)

aliceriot commented:
couple comments, back to you!

roberthouse54 commented:
On the Profile, is it just the spacing at the top (or lack of it?) If so, yeah, I was expecting that to happen I think.

aliceriot commented:
yeah, it used to have space between the navbar and the top of the profile stuff b/c of the margin on `.single-column`.

roberthouse54 commented:
I am aware that some of these styles are still a bit broken, but I think we will need to change some of the html in the js files, and I thought someone else could do that. For example, we are using different html for buttons on the Sign Up pages and the dashboard. Also, there is a div class "profile-progress-controls" which is aligned incorrectly between Sign Up page 1 and Page 2, because on page 1 it is inside a card, and on other pages it is outside the card. So we should just probably move it outside the cards on all pages. There are some other places where things are a bit broken, but I think it is moving in the right direction. I also spoke with George about adding IDs on the page level, so I can define selectors based on which page we are on.

noisecapella commented:
It looks like this PR includes a lot of other outside commits... I'll clean this up

noisecapella commented:
@roberthouse54 To update your branch, do these commands:

    git checkout master
    git pull
    git branch -D rh_css-style-changes
    git checkout rh_css-style-changes

This just updates master, deletes your local branch and creates a fresh copy based on the updated version on GitHub.

noisecapella commented:
Also, you will need to address the comments regarding whitespace that @aliceriot made

aliceriot commented:
@

aliceriot commented:
srry clicked the wrong button >_<

noisecapella commented on static/scss/_button.scss:
```diff
@@ -23,11 +23,12 @@
-  box-shadow: 2px 2px 8px grey;
-  font-weight: 600;
+  font-weight: 400;
+  font-size: 13px;
```
For some reason I had though `pt` was preferable to `px`, but it looks like this is correct: http://stackoverflow.com/questions/3557260/should-i-use-pt-or-px

We should switch over the other font sizes which use `pt` whenever we get around to it

noisecapella commented on static/scss/_variables.scss:
```diff
@@ -1,14 +1,26 @@
 // colors
+
```
Can you remove the empty lines here?

roberthouse54 replied:
All set.

noisecapella commented on static/scss/bound_check_box.scss:
```diff
@@ -13,9 +13,12 @@
       font-size: 14px;
-      color: $font-black;
+      color: $medlightgrey;
+      position: relative;
```
What's this for?

roberthouse54 replied:
This is the text for agreeing to the Terms of Service in the sign up flow. The relative positioning aligns it better with the checkbox image.

noisecapella commented on static/scss/general_styles.scss:
```diff
@@ -0,0 +1,45 @@
+  background-color: $page-background;
+  font-family: 'Roboto', helvetica, arial, sans-serif !important;
+  line-height: 1.4em;
+  }
```
Can you remove the spaces before this `}`?

noisecapella commented on static/scss/dashboard.scss:
```diff
@@ -78,26 +76,19 @@ c.validation-wrapper {
     flex-direction: row;
 
     .dashboard-user-card-text {
+
```
Can you remove this line?

noisecapella commented on static/scss/general_styles.scss:
```diff
@@ -0,0 +1,45 @@
+  }
+
+
+ .mdl-card {
```
Remove the space before `.mdl-card`

noisecapella commented on static/scss/general_styles.scss:
```diff
@@ -0,0 +1,45 @@
+
+
+ .mdl-card {
+    padding: 20px 24px 30px;
```
Indentation should be two spaces per indent, no tabs. I can show you where to set this. Also this may be useful for Sublime, which will make it easy to clean the whitespace up afterwards: https://packagecontrol.io/packages/CleanCSS

noisecapella commented on static/scss/general_styles.scss:
```diff
@@ -0,0 +1,45 @@
+  }
+
+
+.mdl-shadow--2dp, .mdl-shadow--3dp {box-shadow: none;}
```
Can you put each CSS directive on its own line? Ie

    .mdl-shadow--2dp, .mdl-shadow--3dp {
      box-shadow: none;
    }

noisecapella commented on static/scss/navbar.scss:
```diff
@@ -41,6 +45,7 @@
     border-right-color: whte;
     border-top-color: white;
     outline-color: white;
+    a:hover, a:focus {text-decoration: none;}
```
Please put `text-decoration: none;` on its own line

noisecapella commented on static/scss/page_layout.scss:
```diff
@@ -1,25 +1,31 @@
+$border-box-sizing: true !global;
+
+
```
Remove this line and the next one please

noisecapella commented on static/scss/profile.scss:
```diff
@@ -71,6 +72,11 @@
+
+  .bound-check-box {
+    margin:  12px 0 20px;
+
```
Remove this line please

noisecapella commented on static/scss/profile.scss:
```diff
@@ -38,9 +38,9 @@
-    font-size: 9pt;
+    font-weight: 400;
+    font-size: 12px;
+
```
Remove this line please

noisecapella commented on static/scss/progress_widget.scss:
```diff
@@ -35,16 +37,20 @@
-    margin: 0;
+    padding: 0;
+    margin: 0 0 13px;
+    
```
Remove this line please

noisecapella commented on static/scss/user_menu.scss:
```diff
@@ -20,27 +20,34 @@
+    font-weight: 300;
+  }
+
+  .user-menu-dropdown > a:hover {
```
Feel free to leave this but you can also nest these kinds of directives:

    .user-menu-dropdown {
      > a {
        ...
      }
      
      > a:hover {
        ...
      }

noisecapella commented on static/scss/welcome_banner.scss:
```diff
@@ -14,8 +14,9 @@
     padding-left: 25px;
-    max-width: 70%;
-    font-size: 16px;
+    /* max-width: 70%; */
```
Can you delete this line?

aliceriot commented:
@noisecapella are you reviewing this now? i can un-assign myself. Didn't see a 'needs review' tag so I haven't added any comments.

noisecapella commented:
@roberthouse54 asked me to take a look at it but I'm done commenting now

roberthouse54 commented:
Made all the requested changes.

aliceriot commented:
should this have black background?
![usermenu](https://cloud.githubusercontent.com/assets/6207644/17944538/390f7af0-6a0e-11e6-9072-e4a907d1ef6d.png)

aliceriot commented:
the button alignment on the education and work history profile steps is different:

![button-alignment](https://cloud.githubusercontent.com/assets/6207644/17944669/afdc654e-6a0e-11e6-9cf8-fe2fcd3d2af2.png)

aliceriot commented:
also the cards don't have shadows anymore

aliceriot commented:
userchip stretches off the side of the page:

![stretch](https://cloud.githubusercontent.com/assets/6207644/17944756/0cbbd2a4-6a0f-11e6-8966-67ccba765bb6.png)

aliceriot commented:
@roberthouse54 just read your comment above, are the above things I screenshotted all things you were already aware of?

roberthouse54 commented:
The changes are intentional, apart from the user chip one. For some reason I am not seeing any results in my Learner Search page, so I havent looked at that page yet. I removed the shadows from cards, and changed the user dropdown to black. The button alignment on the sign up pages was a bit of a problem. I think it may be best to move that button group outside of the card on page 1, so it was consistent on all pages.

aliceriot commented:
Yeah, I agree re: button movement, it's good to be consistent about things like that!

aliceriot commented on static/scss/general_styles.scss:
```diff
@@ -0,0 +1,47 @@
+    padding: 20px 24px 30px;
+    margin: 0 0 22px 0;
+    min-height: 150px;
+    width: 100%;
```
this is what's causing the issue with the user chip - I think we should remove it.

I took out this line on my machine and it looks like the only things which are affected negatively are the cards on the `/dashboard` - could we add a `full-width` class to those and then scope the `width: 100%` rule to `.mdl-card.full-width`?

aliceriot commented on static/scss/navbar.scss:
```diff
@@ -41,6 +45,9 @@
     outline-color: white;
+    a:hover, a:focus {
+      text-decoration: none;
+      }
```
did you want to get rid of the hover underline on the navbar links?

noisecapella replied:
Can we also unindent this line?

aliceriot commented:
There's two things basically to address:

- code formatting (mostly indentation)
- the userchip thing

aliceriot commented:
back to you!

roberthouse54 commented:
About the user chip. I think I'd rather leave the 100% width on mdl-card as that is the usual behavior for cards. I'd create a special class for the user chip and set the width there.

roberthouse54 commented:
I guess what I mean is that we should assume that cards are full width of whatever container they are in. If a card is not full width we should overwrite that. The userchip also should have a dropshadow, but that is removed from all other cards. The user chip is an exception in other words, it is not a normal card.

aliceriot commented:
Ok yeah, that makes sense to me.

roberthouse54 commented:
Would it be better to merge this as is, and have @aliceriot or @noisecapella make any html changes, or should I update the JS files myself?

aliceriot commented:
Well we should definitely fix the code formatting before merging!

Not sure, @noisecapella what do you think?

noisecapella commented:
I think one of us should take over this PR, then merge after the additional changes are included

aliceriot commented:
@roberthouse54 I just pushed the change to make the name hover - I think we should do the layout change for `LearnerResult` in a separate pr, there's a little bit more to it.

aliceriot commented:
@noisecapella would you mind doing the review from here on out? I wrote some of the code, so it seems like I shouldn't review it, haha

noisecapella commented:
Sure, I just assigned myself

noisecapella commented:
The tests are failing, can you investigate @aliceriot?

aliceriot commented:
yes, I'll take a look.

aliceriot commented:
broken tests were due to selector changes, should be fixed!

noisecapella commented on static/js/components/ProfileProgressControls.js:
```diff
@@ -30,15 +30,15 @@ export default class ProfileProgressControls extends React.Component {
     if(prevStep) {
       prevButton = <button
-        className="mm-button-action prev"
+        className="mm-button action prev"
```
It doesn't look like `action` exists anywhere in the CSS, is it declared by material-design?

aliceriot replied:
No, we had it declared but aren't using anymore, I'll remove it.

noisecapella commented on static/scss/user_menu.scss:
```diff
@@ -20,28 +20,37 @@
+    
+    > a:first-child {
+      margin-top: 12px;
+      }
```
Can each of these be unindented to match with the `>` above?

noisecapella commented on static/scss/profile.scss:
```diff
@@ -71,6 +71,10 @@
+
+  .bound-check-box {
+    margin:  12px 0 20px;
+    }
```
Can you unindent this line to match `.bound-...` above?

noisecapella commented on static/scss/_button.scss:
```diff
@@ -1,35 +1,42 @@
-  cursor: pointer
-}
-
-.mm-button-action {
```
It looks like this is still used in two places, in `ProgressWidget` and `CourseAction`. If we're fine with how they look now we should remove the references to `mm-button-action` there

aliceriot replied:
I went ahead and removed both of those.

aliceriot commented:
@noisecapella CSS code formatting should be sorted.

noisecapella commented:
:+1:

aliceriot commented:
🎆 🎊 💯
